assertion/function/analyzer.go

```diff
@@ -272,13 +272,20 @@ func duplicateFullTriggersFromContractedFunctionsToCallers(
 ) {
 
 	// Find all the calls to contracted functions
-	// callsByCtrtFunc is a mapping contracted function -> caller -> call expression
+	// callsByCtrtFunc is a mapping: contracted function -> caller -> all the call expressions
```

Comment:
How does this work, though? Isn't `*ast.CallExpr{}` a single call expression rather than a list/set of expressions? Even the loop `for caller, callExpr := range calls` below gives one `callExpr` per `caller`, no?

Reply:
You are absolutely right! Nice catch! Fixed.